bootstrap/pkg/kfapp/ksonnet/ksonnet.go

```diff
@@ -288,6 +288,50 @@ func (ksApp *ksApp) deleteGlobalResources(config *rest.Config) error {
 }
 
 func (ksApp *ksApp) Delete(resources kftypes.ResourceEnum) error {
+	if ksApp.Spec.Platform != kftypes.GCP {
```

Comment:
Why aren't you invoking this code on GCP? If you are deleting K8s resources I would expect that to also work on GCP.

On GCP I think we want to delete K8s resources before deleting the cluster because this triggers some cleanup.

e.g. deleting ingress should allow the GCP resources to be cleaned up; whereas if we don't delete the K8s resource and just delete the cluster they won't get cleaned up.

Reply:
what kind of GCP resources?  currently we skip deleting resources for the following: https://github.com/kubeflow/kubeflow/blob/master/testing/test_deploy_app.py#L443-L450

currently these codes won't work if user doesn't point KUBECONFIG to the correct context.

Reply:
Gabriel is right, i think but i’ll verify. I’ll also add an e2e test

Reply:
It looks like this code is deleting all the K8s resources that we created as part of deploying Kubeflow.

So why wouldn't we also delete those resources in the case of GCP platform?

At worst its an unnecessary operation because we will end up deleting the cluster.

However, in some cases if we don't delete the K8s resource and trigger corresponding deletion handlers in K8s we don't properly clean up all resources.

The logic here:
https://github.com/kubeflow/kubeflow/blob/master/testing/test_deploy_app.py#L443-L450

Is indicating that certain resources like http-maps aren't being properly deleted.

We never create those resources explicitly. Those resources are created by the ingress that we create.

When we delete Kubeflow we shouldn't have to explicitly delete those resources (url-map, etc...) The fact that we do is a work around for the fact that those resources aren't being adequately garbage collected when the owning resource (GKE cluster is deleted).

I think if we explicitly delete the K8s resources as opposed to just deleting that cluster helps avoid these types of resource leaks.

Also from a design perspective, its better if we minimize the number of "if ${platform} then ...".

Reply:
@jlewi +1

I also realized that `kfctl delete k8s` would not delete the cluster, so k8 cleanup needs to occur.

>On GCP I think we want to delete K8s resources before deleting the cluster because this triggers some cleanup.

I'll invert the order of operations - (package managers delete -> platforms delete)

Reply:
@jlewi 

For `kfctl delete platform` - it seems like this should behave the same as `kfctl delete all`.

Reply:
I'm on the fence; what do you recommend?

What happens if there is a bug that prevents 

`kfctl delete k8s` from succeeding?

In that case we might want to fall back to just running `kfctl delete platform`.

So maybe the behavior could be

`kfctl delete all` - Invoke k8s delete and abort on error

`kfctl delete platform` - try to invoke k8s delete but continue to platform if k8s delete throws an error.

Thoughts?